diff --git a/core/modules/views/src/Plugin/views/row/RssFields.php b/core/modules/views/src/Plugin/views/row/RssFields.php

--- a/core/modules/views/src/Plugin/views/row/RssFields.php
+++ b/core/modules/views/src/Plugin/views/row/RssFields.php
@@ -15,2 +15,3 @@ namespace Drupal\views\Plugin\views\row;
use Drupal\image\Entity\ImageStyle;
use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\Component\Utility\UrlHelper;
 
@@ -194,7 +194,15 @@ public function render($row) {
     // Create the RSS item object.
     $item = new \stdClass();
     $item->title = $this->getField($row_index, $this->options['title_field']);
-    $item->link = $this->getAbsoluteUrl($this->getField($row_index, $this->options['link_field']));
+
+    // If internal link, get absolute URL from URI.
+    $link = $this->getField($row_index, $this->options['link_field']);
+    if (!UrlHelper::isExternal($link)) {
+      $item->link = $this->getAbsoluteUrl($link);
+    }
+    else {
+      $item->link = Url::fromUri($this->getField($row_index, $this->options['link_field']))->setAbsolute()->toString();
+    }
 
     $field = $this->getField($row_index, $this->options['description_field']);
     $item->description = is_array($field) ? $field : ['#markup' => $field];
@@ -281,7 +281,14 @@ public function render($row) {
     $item_guid = $this->getField($row_index, $this->options['guid_field_options']['guid_field']);
     if ($this->options['guid_field_options']['guid_field_is_permalink']) {
       $guid_is_permalink_string = 'true';
-      $item_guid = $this->getAbsoluteUrl($item_guid);
+
+      // If the guid is an internal link, get the absolute URL from the URI.
+      if (!UrlHelper::isExternal($item_guid)) {
+        $item_guid = $this->getAbsoluteUrl($item_guid);
+      }
+      else {
+        $item_guid = Url::fromUri($item_guid);
+      }
     }
     $item->elements[] = [
       'key' => 'guid',
