diff --git a/core/modules/filter/src/Plugin/Filter/FilterAlign.php b/core/modules/filter/src/Plugin/Filter/FilterAlign.php
index a3739ab824..1ef59b2c37 100644
--- a/core/modules/filter/src/Plugin/Filter/FilterAlign.php
+++ b/core/modules/filter/src/Plugin/Filter/FilterAlign.php
@@ -31,14 +31,16 @@ public function process($text, $langcode) {
         // Read the data-align attribute's value, then delete it.
         $align = $node->getAttribute('data-align');
         $node->removeAttribute('data-align');
-
+        // UCSF - Remove restriction on only left, center, right
+        $classes = $node->getAttribute('class');
+        $classes = (strlen($classes) > 0) ? explode(' ', $classes) : [];
         // If one of the allowed alignments, add the corresponding class.
         if (in_array($align, ['left', 'center', 'right'])) {
-          $classes = $node->getAttribute('class');
-          $classes = (strlen($classes) > 0) ? explode(' ', $classes) : [];
           $classes[] = 'align-' . $align;
-          $node->setAttribute('class', implode(' ', $classes));
+        } else {
+          $classes[] = $align;
         }
+        $node->setAttribute('class', implode(' ', $classes));
       }
       $result->setProcessedText(Html::serialize($dom));
     }
