diff --git a/core/modules/filter/src/Plugin/Filter/FilterCaption.php b/core/modules/filter/src/Plugin/Filter/FilterCaption.php
index 817f73b7a..4fbe6196e 100644
--- a/core/modules/filter/src/Plugin/Filter/FilterCaption.php
+++ b/core/modules/filter/src/Plugin/Filter/FilterCaption.php
@@ -71,7 +71,7 @@ public function process($text, $langcode) {
       $xpath = new \DOMXPath($dom);
       $html_filter = $this->filterManager->createInstance('filter_html', [
         'settings' => [
-          'allowed_html' => '<a href hreflang target rel> <em> <strong> <cite> <code> <br>',
+          'allowed_html' => '<a href hreflang target rel> <em> <strong> <cite> <code> <br> <p>',
           'filter_html_help' => FALSE,
           'filter_html_nofollow' => FALSE,
         ],
@@ -83,7 +83,7 @@ public function process($text, $langcode) {

         // Sanitize caption: decode HTML encoding, limit allowed HTML tags; only
         // allow inline tags that are allowed by default, plus <br>.
-        $caption = Html::decodeEntities($caption);
+        $caption = Html::decodeEntities(Html::decodeEntities($caption));
         $raw_caption = $caption;
         $filtered_caption = $html_filter->process($caption, $langcode);
         $result->addCacheableDependency($filtered_caption);
