--- a/entity_embed_link.module
+++ b/entity_embed_link.module
@@ -62,7 +62,7 @@
  * Implements hook_theme_registry_alter().
  */
 function entity_embed_link_theme_registry_alter(&$theme_registry) {
-  $theme_registry['entity_embed_container']['path'] = \Drupal::service('extension.list.module')->getPath('entity_embed_link') . '/templates';
+  //$theme_registry['entity_embed_container']['path'] = \Drupal::service('extension.list.module')->getPath('entity_embed_link') . '/templates';
 }
 
 /**

