diff --git a/entity_embed_link.module b/entity_embed_link.module
index dca4b907e..c9c040b7c 100644
--- a/entity_embed_link.module
+++ b/entity_embed_link.module
@@ -38,7 +38,7 @@ function entity_embed_link_form_entity_embed_dialog_alter(&$form, FormStateInter
  * Implements hook_theme_registry_alter()
  */
 function entity_embed_link_theme_registry_alter(&$theme_registry) {
-  $theme_registry['entity_embed_container']['path'] = drupal_get_path('module', 'entity_embed_link') . '/templates';
+ //$theme_registry['entity_embed_container']['path'] = drupal_get_path('module', 'entity_embed_link') . '/templates';
 }
 
 /**
