diff --git a/js/plugins/drupalentity/plugin.js b/js/plugins/drupalentity/plugin.js
index b8a5090c9..578a7841a 100644
--- a/js/plugins/drupalentity/plugin.js
+++ b/js/plugins/drupalentity/plugin.js
@@ -62,6 +62,11 @@
             var entityElement = editor.document.createElement('drupal-entity');
             var attributes = values.attributes;
             for (var key in attributes) {
+              if (key == 'data-caption') {
+                if (typeof attributes[key] === 'object') {
+                  attributes[key] = attributes[key].value;
+                }
+              }
               entityElement.setAttribute(key, attributes[key]);
             }
 
diff --git a/src/Form/EntityEmbedDialog.php b/src/Form/EntityEmbedDialog.php
index ae0194b38..70e876385 100644
--- a/src/Form/EntityEmbedDialog.php
+++ b/src/Form/EntityEmbedDialog.php
@@ -445,12 +445,15 @@ public function buildEmbedStep(array $form, FormStateInterface $form_state) {
     if ($editor->getFilterFormat()->filters('filter_align')->status) {
       $form['attributes']['data-align'] = array(
         '#title' => $this->t('Align'),
-        '#type' => 'radios',
+        '#type' => 'select',
         '#options' => array(
           '' => $this->t('None'),
           'left' => $this->t('Left'),
           'center' => $this->t('Center'),
           'right' => $this->t('Right'),
+          'bleed-left' => $this->t('Bleed-Left'),
+          'bleed-right' => $this->t('Bleed-Right'),
+          'bleed-full' => $this->t('Bleed-Full'),
         ),
         '#default_value' => isset($entity_element['data-align']) ? $entity_element['data-align'] : '',
         '#wrapper_attributes' => array('class' => array('container-inline')),
@@ -463,7 +466,7 @@ public function buildEmbedStep(array $form, FormStateInterface $form_state) {
     if ($editor->getFilterFormat()->filters('filter_caption')->status) {
       $form['attributes']['data-caption'] = array(
         '#title' => $this->t('Caption'),
-        '#type' => 'textfield',
+        '#type' => 'text_format', //textfield or text_format
         '#default_value' => isset($entity_element['data-caption']) ? Html::decodeEntities($entity_element['data-caption']) : '',
         '#element_validate' => array('::escapeValue'),
       );
@@ -451,9 +451,11 @@ public function buildEmbedStep(array $form, FormStateInterface $form_state) {
          'left' => $this->t('Left'),
          'center' => $this->t('Center'),
          'right' => $this->t('Right'),
-         'bleed-left' => $this->t('Bleed-Left'),
-         'bleed-right' => $this->t('Bleed-Right'),
-         'bleed-full' => $this->t('Bleed-Full'),
+         'half-image-right' => $this->t('Half Image Right'),
+         'half-image-left' => $this->t('Half Image Left'),
+         'half-image-right-full' => $this->t('Full Bleed Right'),
+         'half-image-left-full' => $this->t('Full Bleed Left'),
+         'full-bleed-image' => $this->t('Full Bleed')
        ),
        '#default_value' => isset($entity_element['data-align']) ? $entity_element['data-align'] : '',
        '#wrapper_attributes' => array('class' => array('container-inline')),
 
