diff --git a/docroot/modules/contrib/metatag/src/Plugin/metatag/Tag/MetaNameBase.php b/docroot/modules/contrib/metatag/src/Plugin/metatag/Tag/MetaNameBase.php
index 9212571ef..c3bddb2f4 100644
--- a/docroot/modules/contrib/metatag/src/Plugin/metatag/Tag/MetaNameBase.php
+++ b/docroot/modules/contrib/metatag/src/Plugin/metatag/Tag/MetaNameBase.php
@@ -325,6 +325,10 @@ public function output() {
     $elements = [];
     foreach ($values as $value) {
       $value = $this->tidy($value);
+      // Validate if the value is empty here, otherwise we get absolute URL.
+      if (empty($value)) {
+        continue;
+      }
       if ($this->requiresAbsoluteUrl()) {
         // Relative URL.
         if (parse_url($value, PHP_URL_HOST) == NULL) {
