diff --git a/metatag_open_graph/src/Plugin/metatag/Tag/OgDescription.php b/metatag_open_graph/src/Plugin/metatag/Tag/OgDescription.php
index 8cad733c0..570b999e0 100644
--- a/metatag_open_graph/src/Plugin/metatag/Tag/OgDescription.php
+++ b/metatag_open_graph/src/Plugin/metatag/Tag/OgDescription.php
@@ -16,7 +16,7 @@
  *   weight = 6,
  *   type = "label",
  *   secure = FALSE,
- *   multiple = FALSE
+ *   multiple = TRUE
  * )
  */
 class OgDescription extends MetaPropertyBase {
diff --git a/metatag_open_graph/src/Plugin/metatag/Tag/OgTitle.php b/metatag_open_graph/src/Plugin/metatag/Tag/OgTitle.php
index 8929887bf..4ffc36b55 100644
--- a/metatag_open_graph/src/Plugin/metatag/Tag/OgTitle.php
+++ b/metatag_open_graph/src/Plugin/metatag/Tag/OgTitle.php
@@ -16,7 +16,7 @@
  *   weight = 4,
  *   type = "label",
  *   secure = FALSE,
- *   multiple = FALSE
+ *   multiple = TRUE
  * )
  */
 class OgTitle extends MetaPropertyBase {
diff --git a/metatag_twitter_cards/src/Plugin/metatag/Tag/TwitterCardsDescription.php b/metatag_twitter_cards/src/Plugin/metatag/Tag/TwitterCardsDescription.php
index eb3cdbb4b..0d5d67ed1 100644
--- a/metatag_twitter_cards/src/Plugin/metatag/Tag/TwitterCardsDescription.php
+++ b/metatag_twitter_cards/src/Plugin/metatag/Tag/TwitterCardsDescription.php
@@ -16,7 +16,7 @@
  *   weight = 2,
  *   type = "label",
  *   secure = FALSE,
- *   multiple = FALSE
+ *   multiple = TRUE
  * )
  */
 class TwitterCardsDescription extends MetaNameBase {
diff --git a/metatag_twitter_cards/src/Plugin/metatag/Tag/TwitterCardsImage.php b/metatag_twitter_cards/src/Plugin/metatag/Tag/TwitterCardsImage.php
index 4fa71b703..c364541d0 100644
--- a/metatag_twitter_cards/src/Plugin/metatag/Tag/TwitterCardsImage.php
+++ b/metatag_twitter_cards/src/Plugin/metatag/Tag/TwitterCardsImage.php
@@ -16,7 +16,7 @@
  *   weight = 7,
  *   type = "image",
  *   secure = FALSE,
- *   multiple = FALSE,
+ *   multiple = TRUE,
  *   absolute_url = TRUE
  * )
  */
diff --git a/metatag_twitter_cards/src/Plugin/metatag/Tag/TwitterCardsTitle.php b/metatag_twitter_cards/src/Plugin/metatag/Tag/TwitterCardsTitle.php
index 268dfa4bd..45be445e5 100644
--- a/metatag_twitter_cards/src/Plugin/metatag/Tag/TwitterCardsTitle.php
+++ b/metatag_twitter_cards/src/Plugin/metatag/Tag/TwitterCardsTitle.php
@@ -16,7 +16,7 @@
  *   weight = 2,
  *   type = "label",
  *   secure = FALSE,
- *   multiple = FALSE
+ *   multiple = TRUE
  * )
  */
 class TwitterCardsTitle extends MetaNameBase {
diff --git a/src/MetatagManager.php b/src/MetatagManager.php
index 57c4ca281..ba17a7844 100644
--- a/src/MetatagManager.php
+++ b/src/MetatagManager.php
@@ -552,8 +552,23 @@ public function generateRawElements(array $tags, $entity = NULL) {
           }
 
           foreach ($output as $index => $element) {
-            // Add index to tag name as suffix to avoid having same key.
-            $index_tag_name = $tag->multiple() ? $tag_name . '_' . $index : $tag_name;
+            // UCSFD8-323 - only display the first valid token from left to right order in the Metatag field(s)
+            $single_tags = ['og_title','og_description','og_image','twitter_cards_title','twitter_cards_description','twitter_cards_image'];
+            // String elements might be empty, so we need to skip over those
+            if (in_array($tag_name, $single_tags)) {
+              if (trim($element['#attributes']['content'])=='') {
+                continue;
+              }
+              // Check if raw tag is created there should only be one value
+              if (isset($rawTags[$tag_name])) {
+                continue;
+              }
+              // Create a tag with no index appendended
+              $index_tag_name = $tag_name;
+            } else {
+              // Add index to tag name as suffix to avoid having same key.
+              $index_tag_name = $tag->multiple() ? $tag_name . '_' . $index : $tag_name;
+            }
             $rawTags[$index_tag_name] = $element;
           }
         }
diff --git a/src/Plugin/metatag/Tag/MetaNameBase.php b/src/Plugin/metatag/Tag/MetaNameBase.php
index 9212571ef..ac32bcbcd 100644
--- a/src/Plugin/metatag/Tag/MetaNameBase.php
+++ b/src/Plugin/metatag/Tag/MetaNameBase.php
@@ -321,10 +321,14 @@ public function output() {
 
     // Parse out the image URL, if needed.
     $value = $this->parseImageUrl();
-    $values = $this->multiple() ? explode(',', $value) : [$value];
+    $values = $this->multiple() ? explode('||', $value) : [$value];
     $elements = [];
     foreach ($values as $value) {
       $value = $this->tidy($value);
+      // Validate if the value is empty here, otherwise we get absolute URL.
+      if (empty($value)) {
+        continue;
+      }
       if ($this->requiresAbsoluteUrl()) {
         // Relative URL.
         if (parse_url($value, PHP_URL_HOST) == NULL) {
