diff --git a/metatag_twitter_cards/src/Plugin/metatag/Tag/TwitterCardsImage.php b/metatag_twitter_cards/src/Plugin/metatag/Tag/TwitterCardsImage.php
index 4fa71b703..c364541d0 100644
--- a/metatag_twitter_cards/src/Plugin/metatag/Tag/TwitterCardsImage.php
+++ b/metatag_twitter_cards/src/Plugin/metatag/Tag/TwitterCardsImage.php
@@ -16,7 +16,7 @@
  *   weight = 7,
  *   type = "image",
  *   secure = FALSE,
- *   multiple = FALSE,
+ *   multiple = TRUE,
  *   absolute_url = TRUE
  * )
  */
