diff --git a/src/Plugin/video_embed_field/Provider/Vimeo.php b/src/Plugin/video_embed_field/Provider/Vimeo.php
index 70c7f6beb..701dd468b 100644
--- a/src/Plugin/video_embed_field/Provider/Vimeo.php
+++ b/src/Plugin/video_embed_field/Provider/Vimeo.php
@@ -18,10 +18,25 @@ class Vimeo extends ProviderPluginBase {
    * {@inheritdoc}
    */
   public function renderEmbedCode($width, $height, $autoplay) {
+    /*
+    * Modification to allow the ability to add 1 option to the end of a vimeo embed
+    * Get the raw input of what was entered for the url.
+    * Match to get the option after the url
+    * Conmbine it with the videoId string to append it to iframe #url item
+    */
+    $input = $this->getInput();
+    preg_match('/^https?:\/\/(www\.)?vimeo.com\/(channels\/[a-zA-Z0-9]*\/)?(?<id>[0-9]*)(\/[a-zA-Z0-9]+)?(\#t=(\d+)s)?(\?((?<option>[a-zA-Z0-9]*=[a-zA-Z0-9]*)))?$/',$input ,$option );
+    if($option['option']){
+      $url = $this->getVideoId(). "?".$option['option']."&";
+    }
+    else {
+      $url = $this->getVideoId();
+    }
+
     $iframe = [
       '#type' => 'video_embed_iframe',
       '#provider' => 'vimeo',
-      '#url' => sprintf('https://player.vimeo.com/video/%s', $this->getVideoId()),
+      '#url' => sprintf('https://player.vimeo.com/video/%s', $url),
       '#query' => [
         'autoplay' => $autoplay,
       ],
@@ -59,7 +74,7 @@ protected function oEmbedData() {
    * {@inheritdoc}
    */
   public static function getIdFromInput($input) {
-    preg_match('/^https?:\/\/(www\.)?vimeo.com\/(channels\/[a-zA-Z0-9]*\/)?(?<id>[0-9]*)(\/[a-zA-Z0-9]+)?(\#t=(\d+)s)?$/', $input, $matches);
+    preg_match('/^https?:\/\/(www\.)?vimeo.com\/(channels\/[a-zA-Z0-9]*\/)?(?<id>[0-9]*)(\/[a-zA-Z0-9]+)?(\#t=(\d+)s)?(\?background=(?<option>[a-zA-Z0-9]*))?$/', $input, $matches);
     return isset($matches['id']) ? $matches['id'] : FALSE;
   }
 
